### PR TITLE
add global breadcrumbs for audit log pages

### PR DESCRIPTION
**Pull request by tianleh**
…t logging

*Issue #, if available:*

*Description of changes:*
This change is to add breadcrumbs display for audit logging pages. 

After checking the existing behavior of the code, it will only add breadcrumbs to the top of the right-side page if there is nav panel instead of treating nav panel + the right-side page as a whole.

I tried to follow the current logic for navpanel to add another Route for breadcrumbs. However, there are a few problems:
1) app-route becomes more complicated with branching logic. If only navpanel, it looks fine. But with another component, it starts becoming messy.
2) Within audit logging, there are a few pages whose breadcrumbs title needs to be update. E.g general vs compliance. It seems that the app-route only uses the initial values without dynamically changing during same-hash pages.

Thus I decide to make some refactoring so that each page takes care of their own rendering of side panel and breadcrumbs. The benefit is that we only need to look at the feature's own tsx to understand its rendering effect and keep app-route simple.

*Test:*
Verify that all the pages this PR touch have the same display for nav panel and breadcrumbs as before.

Specifically for audit logging,
Below are the mockups

screencapture-aws-uxdr-invisionapp-share-ZTVZTJJKURY-2020-07-23-23_20_45

screencapture-aws-uxdr-invisionapp-share-ZTVZTJJKURY-2020-07-23-23_21_03

screencapture-aws-uxdr-invisionapp-share-ZTVZTJJKURY-2020-07-23-23_21_35

Implementation:
screencapture-localhost-5601-app-opendistro-security-2020-07-23-23_22_26

screencapture-localhost-5601-app-opendistro-security-2020-07-23-23_22_45

screencapture-localhost-5601-app-opendistro-security-2020-07-23-23_23_04

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/apps/configuration/utils/display-utils.tsx [hunk 1]
```diff
@@ -51,3 +55,48 @@ export function renderCustomization(reserved: boolean) {
...
+      breadcrumbs={[
+        {
+          text: 'Security',
+          href: buildHashUrl(),
+        },
+        {
+          text: ROUTE_MAP[resourceType].name,
+          href: buildHashUrl(resourceType),
+        },
+        {
```
Comment (yuxitang-amzn) [inline public/apps/configuration/utils/display-utils.tsx:96]: These 2 routes looks quite same. Suggest we define the same part as a base value, and concat the third one if title provided

**Comment by yuxitang-amzn** [comment]
Comment for design:
I really doubt why need both breadcrumbs and nav panel on same page. They are both for navigation purpose and adding one more does not provide more functions. It doesn't provide more value to the user but makes code more complicated.

And if want to set breadcrumbs like the UI mockup, need to use the appMountContext to set instead of a component. See:
https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/commit/fc24a532a36e406ec8ee346a80746367ad39ae05#diff-060a44670bd638f9195592c8cfccf9ecR38-R43

**Comment by yuxitang-amzn** [comment]
In my opinion, ideally each page except router should be a pure page, they better not be aware of the path and url. While putting a breadcrumb into the page kind of breaks that.

File: public/apps/configuration/utils/display-utils.tsx (hunk 1 above)
Comment (tianleh) [reply public/apps/configuration/utils/display-utils.tsx]: True. Switched to concat

**Comment by tianleh** [comment]
> In my opinion, ideally each page except router should be a pure page, they better not be aware of the path and url. While putting a breadcrumb into the page kind of breaks that.

Reverted previous changes and add breadcrumbs in render function inside Route.

**Comment by tianleh** [comment]
> Comment for design:
> I really doubt why need both breadcrumbs and nav panel on same page. They are both for navigation purpose and adding one more does not provide more functions. It doesn't provide more value to the user but makes code more complicated.
> 
> And if want to set breadcrumbs like the UI mockup, need to use the appMountContext to set instead of a component. See:
> [fc24a53#diff-060a44670bd638f9195592c8cfccf9ecR38-R43](https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/commit/fc24a532a36e406ec8ee346a80746367ad39ae05#diff-060a44670bd638f9195592c8cfccf9ecR38-R43)

Confirmed with UX. Breadcrumbs provides more context, e.g general vs compliance setting which are both under audit logging panel.

**Comment by tianleh** [comment]
Below are the screenshots from the updated implementation.

screencapture-localhost-5601-app-opendistro-security-2020-07-24-20_03_00

screencapture-localhost-5601-app-opendistro-security-2020-07-24-20_03_18

screencapture-localhost-5601-app-opendistro-security-2020-07-24-20_03_41

**Pull request merged**